nmoscommon/auth/nmos_auth.py

```diff
@@ -42,83 +43,109 @@
 APINAMESPACE = "x-nmos"
 APINAME = "auth"
 APIVERSION = "v1.0"
-CERT_ENDPOINT = 'certs'
-CERT_URL_PATH = '/{}/{}/{}/{}'.format(APINAMESPACE, APINAME, APIVERSION, CERT_ENDPOINT)
+JWK_ENDPOINT = 'jwks'
+JWK_URL_PATH = '/{}/{}/{}/{}'.format(APINAMESPACE, APINAME, APIVERSION, JWK_ENDPOINT)
 
 
 class RequiresAuth(object):
 
-    def __init__(self, condition=OAUTH_MODE, claimsOptions=IS_XX_CLAIMS,
-                 certificate=None):
+    def __init__(self, condition=OAUTH_MODE, claimsOptions=IS_XX_CLAIMS):
+        self.logger = defaultLogger("authresource")
+        self.bridge = IppmDNSBridge()
         self.condition = condition
         self.claimsOptions = claimsOptions
-        self.certificate = certificate
-        self.bridge = IppmDNSBridge()
-        self.logger = defaultLogger("authresource")
+        self.publicKey = None
 
     def getHrefFromService(self, serviceType):
         return self.bridge.getHref(serviceType)
 
-    def getCertFromEndpoint(self):
+    def getJwksFromEndpoint(self):
         try:
             href = self.getHrefFromService(MDNS_SERVICE_TYPE)
-            certHref = urljoin(href, CERT_URL_PATH)
-            self.logger.writeInfo('cert href is: {}'.format(certHref))
-            cert_resp = requests.get(certHref, timeout=0.5, proxies={'http': ''})
-            cert_resp.raise_for_status()  # Raise error if status !=200
+            jwk_href = urljoin(href, JWK_URL_PATH)
+            self.logger.writeInfo('JWK endpoint is: {}'.format(jwk_href))
+            jwk_resp = requests.get(jwk_href, timeout=0.5, proxies={'http': ''})
+            jwk_resp.raise_for_status()  # Raise error if status !=200
         except RequestException as e:
             self.logger.writeError("Error: {0!s}".format(e))
-            self.logger.writeError("Cannot find certificate at {}. Is the Auth Server Running?".format(certHref))
+            self.logger.writeError("Cannot find certificate at {}. Is the Auth Server Running?".format(jwk_href))
             raise
 
-        contentType = cert_resp.headers['content-type'].split(";")[0]
-        if contentType == "application/json":
-            cert = cert_resp.json()
+        if "application/json" in jwk_resp.headers['content-type']:
             try:
-                if len(cert) > 1:
-                    self.logger.writeWarning("Multiple certificates at Endpoint. Returning First Instance.")
-                cert = cert[0]
-                return cert
+                jwks = jwk_resp.json()
+                if "keys" in jwks:
```

Comment:
I'd appreciate this being updated to look for the most recent NMOS key as-per the IS-10 draft before we merge it. Otherwise there's a significant danger of picking the wrong key if we use an Auth Server which is managing more than just NMOS-related OAuth.